cpp/src/arrow/util/hash_util.h

```diff
@@ -27,39 +27,27 @@
 
 #include "arrow/util/logging.h"
 #include "arrow/util/macros.h"
-#include "arrow/util/neon_util.h"
-#include "arrow/util/sse_util.h"
+#include "arrow/util/simd.h"
 
-static inline uint32_t HW_crc32_u8(uint32_t crc, uint8_t v) {
-  DCHECK(false) << "Hardware CRC support is not enabled";
-  return 0;
-}
-
-static inline uint32_t HW_crc32_u16(uint32_t crc, uint16_t v) {
-  DCHECK(false) << "Hardware CRC support is not enabled";
-  return 0;
-}
-
-static inline uint32_t HW_crc32_u32(uint32_t crc, uint32_t v) {
-  DCHECK(false) << "Hardware CRC support is not enabled";
-  return 0;
-}
-
-static inline uint32_t HW_crc32_u64(uint32_t crc, uint64_t v) {
+#ifdef ARROW_HAVE_SSE4_2
```

Comment:
How about moving this part into `simd.h`?  We can reduce the number of places that depends on the target hardware.

Reply:
Also, regardless of whether we move them or not, you should put them in the `arrow::internal` namespace.

Reply:
I'd be concerned about putting too much in SIMD directly.  If we expect these to be used in other places that don't depend on this header we should create a specific header targetted at hardware. digests/hashing.